beacon-chain/sync/rpc_beacon_blocks_by_range.go

```diff
@@ -26,181 +26,135 @@ func (s *Service) beaconBlocksByRangeRPCHandler(ctx context.Context, msg interfa
-		tracing.AnnotateError(span, err)
-		return err
+		return rangeParams{}, p2ptypes.ErrInvalidRequest
 	}
-	// handle genesis case
-	if startSlot == 0 {
-		genBlock, genRoot, err := s.retrieveGenesisBlock(ctx)
-		if err != nil {
-			log.WithError(err).Debug("Could not retrieve genesis block")
-			s.writeErrorResponseToStream(responseCodeServerError, p2ptypes.ErrGeneric.Error(), stream)
-			tracing.AnnotateError(span, err)
-			return err
-		}
-		blks = append([]interfaces.ReadOnlySignedBeaconBlock{genBlock}, blks...)
-		roots = append([][32]byte{genRoot}, roots...)
+	if rp.start > maxStart {
+		return rangeParams{}, p2ptypes.ErrInvalidRequest
 	}
-	// Filter and sort our retrieved blocks, so that
-	// we only return valid sets of blocks.
-	blks, roots, err = s.dedupBlocksAndRoots(blks, roots)
+	rp.end, err = rp.start.SafeAdd((rp.size - 1))
 	if err != nil {
-		s.writeErrorResponseToStream(responseCodeServerError, p2ptypes.ErrGeneric.Error(), stream)
-		tracing.AnnotateError(span, err)
-		return err
+		return rangeParams{}, p2ptypes.ErrInvalidRequest
 	}
-	blks, roots = s.sortBlocksAndRoots(blks, roots)
 
-	blks, err = s.filterBlocks(ctx, blks, roots, prevRoot, step, startSlot)
-	if err != nil && err != p2ptypes.ErrInvalidParent {
-		s.writeErrorResponseToStream(responseCodeServerError, p2ptypes.ErrGeneric.Error(), stream)
-		tracing.AnnotateError(span, err)
-		return err
+	limit := uint64(flags.Get().BlockBatchLimit)
+	if limit > maxRequest {
+		limit = maxRequest
 	}
-	start := time.Now()
-	// If the blocks are blinded, we reconstruct the full block via the execution client.
-	blindedExists := false
-	blindedIndex := 0
-	for i, b := range blks {
-		// Since the blocks are sorted in ascending order, we assume that the following
-		// blocks from the first blinded block are also ascending.
-		if b.IsBlinded() {
-			blindedExists = true
-			blindedIndex = i
-			break
-		}
+	if rp.size > limit {
+		rp.size = limit
 	}
 
-	var reconstructedBlock []interfaces.SignedBeaconBlock
-	if blindedExists {
-		reconstructedBlock, err = s.cfg.executionPayloadReconstructor.ReconstructFullBellatrixBlockBatch(ctx, blks[blindedIndex:])
-		if err != nil {
-			log.WithError(err).Error("Could not reconstruct full bellatrix block batch from blinded bodies")
-			s.writeErrorResponseToStream(responseCodeServerError, p2ptypes.ErrGeneric.Error(), stream)
-			return err
-		}
-	}
+	return rp, nil
+}
 
-	for _, b := range blks {
+func (s *Service) writeBlockBatchToStream(ctx context.Context, batch blockBatch, stream libp2pcore.Stream) error {
+	ctx, span := trace.StartSpan(ctx, "sync.WriteBlockRangeToStream")
+	defer span.End()
+
+	blinded := make([]interfaces.ReadOnlySignedBeaconBlock, 0)
+	for _, b := range batch.canonical() {
 		if err := blocks.BeaconBlockIsNil(b); err != nil {
 			continue
 		}
+		checkr, err := b.Block().HashTreeRoot()
+		if err != nil {
+			return err
+		}
+		if b.Root() != checkr {
```

Comment:
This should be impossible, correct ? I am a bit wary about hashing blocks here as it will take be expensive on the CPU for the serving node if they are serving many peers with many batches

Reply:
good catch, I think this must have been some debug code. this has been removed.